src/Middleware/OutputCaching/src/OutputCacheKeyProvider.cs

```diff
@@ -70,7 +68,7 @@ public string CreateStorageKey(OutputCacheContext context)
 
     public static bool ContainsDelimiters(string? value)
     {
-        return !string.IsNullOrEmpty(value) && value.IndexOfAny(KeyDelimiters) > -1;
+        return !string.IsNullOrEmpty(value) && value.AsSpan().IndexOfAny(KeyDelimiter, KeySubDelimiter) > -1;
```

Comment:
```suggestion
        return !string.IsNullOrEmpty(value) && value.AsSpan().IndexOfAny(KeyDelimiter, KeySubDelimiter) >= 0;
```
Is a tiny little bit better at cpu-level, as the comparison and the jump can be fused into one actual executed instruction.